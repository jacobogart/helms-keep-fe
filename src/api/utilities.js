export const host = 'http://localhost:5000/api/';

export const mockTask = {
  id: '1234',
  isCompleted: false, 
  message: 'Clean the kitchen'
};

export const mockCompletedTask = {
  id: '5678',
  isCompleted: true,
  message: 'Make a mock completed task'
}

export const mockQuest = {
	id: '1234',
  title: 'My First Quest',
  color: "green",
	challenges: [ mockTask, mockCompletedTask ]
};
