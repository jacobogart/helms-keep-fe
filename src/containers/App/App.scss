.App {
  position: relative;
}

.background {
  background: url('https://i.imgur.com/IOlKfQy.png');
  background-size: cover;
  height: 100vh;
  width: 100vw;
  position: fixed;
}

button {
  background: $btnColor;

  &:hover {
    background: $btnHoverColor;
  }
}

.dialog-overlay {
  align-items: center;
  background: rgba(0, 0, 0, 0.8);
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  z-index: 3;
}
