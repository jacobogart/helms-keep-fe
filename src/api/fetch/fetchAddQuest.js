import { host } from '../utilities';
import shortid from 'shortid';

export const fetchAddQuest = (title, challenges) => {
	const url = `${host}quests`;
	const body = { id: shortid.generate(), color: 'green', title, challenges };
	const options = {
		method: 'POST',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(body)
	};
	return fetch(url, options).then(response => {
		if (!response.ok) {
			throw Error('Failed to add quest');
		} else {
			return response.json();
		}
	});
};
