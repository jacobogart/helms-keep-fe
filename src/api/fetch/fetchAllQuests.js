import { host } from '../utilities';

export const fetchAllQuests = () => {
	return fetch(`${host}quests`).then(response => {
		if (!response.ok) {
			throw Error('Failed to get quests');
		} else {
			return response.json();
		}
	});
};
