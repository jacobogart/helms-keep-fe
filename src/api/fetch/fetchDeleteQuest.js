import { host } from '../utilities';

export const fetchDeleteQuest = id => {
	const url = `${host}quests/${id}`;
	const init = {
		method: 'DELETE',
		headers: { 'Content-Type': 'application/json' }
	};
	return fetch(url, init).then(response => {
		if (!response.ok) {
			throw Error('Failed to delete quest');
		} else {
			return response;
		}
	});
};
