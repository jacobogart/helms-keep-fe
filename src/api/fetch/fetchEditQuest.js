import { host } from '../utilities';

export const fetchEditQuest = quest => {
	const url = `${host}quests/${quest.id}`;
	const options = {
		method: 'PUT',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(quest)
	};
	return fetch(url, options).then(response => {
		if (!response.ok) {
			throw Error('Failed to edit quest');
		} else {
			return response.json();
		}
	});
};
