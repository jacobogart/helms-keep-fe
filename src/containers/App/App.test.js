import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from './App';
import { mockQuest } from '../../api/utilities';
import { fetchAllQuests } from '../../api/fetch/fetchAllQuests';

jest.mock('../../api/fetch/fetchAllQuests');

describe('App', () => {
  let wrapper;
  const mockStoreQuests = jest.fn();

  fetchAllQuests.mockImplementation(() => Promise.resolve([mockQuest, mockQuest]));
  beforeEach(() => {
    wrapper = shallow(
      <App 
        quests={[mockQuest, mockQuest]} 
        storeQuests={mockStoreQuests}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('CDM', () => {
    it('should call fetchAllQuests if there are no quests', () => {
      wrapper = shallow(<App quests={[]} storeQuests={mockStoreQuests} />);
      expect(fetchAllQuests).toHaveBeenCalled();
    });

    it('should call storeQuests once resolve', async () => {
      wrapper = shallow(<App quests={[]} storeQuests={mockStoreQuests} />);
      await wrapper.instance().componentDidMount()
      expect(mockStoreQuests).toHaveBeenCalledWith([mockQuest, mockQuest])
    });
  });

  describe('mapStateToProps', () => {
    it('should return an away of quests', () => {
      const state = { quests: [mockQuest, mockQuest], test: 'test' };
      const expected = { quests: [mockQuest, mockQuest] };
      const result = mapStateToProps(state);
      expect(result).toEqual(expected);
    });
  });

  // describe('mapDispatchToProps', () => {
  //   it('should call dispatch when storeQuests is called', () => {
  //     const 
  //   });
  // });
});