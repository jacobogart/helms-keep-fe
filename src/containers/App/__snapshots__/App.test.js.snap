// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="background"
  >
    <Header />
    <ConnectFunction />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Main)",
          "type": [Function],
        }
      }
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/new-quest"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/quests/:id"
    />
  </div>
</div>
`;
